### PR TITLE
Add support for HDF4, GEOS, proj4 and xerces-c to GDAL build

### PR DESCRIPTION
**Pull request by gillins**
This PR changes the GDAL build recipe to include the recently added software to make it a more complete install.

It also fixes a problem with the HDF4 recipe for Windows where netcdf was getting included (see https://trac.osgeo.org/gdal/wiki/HDF) potentially causing conflicts. This was already addressed under Linux/Mac.

### CODE REVIEW

**Comment by groutr** [comment]
I was just about to start working on a PR for this. You beat me to it :+1:

**Comment by groutr** [comment]
Could you add some tests that will check for the presence of the new features?

File: gdal/meta.yaml [hunk 1]
```diff
@@ -12,20 +12,28 @@ source:
...
 build:
-  number: 0
+  number: 1
 
 requirements:
   build:
     - python
     - libgdal
     - numpy
+    - hdf4
```
Comment (groutr) [inline gdal/meta.yaml]: Does gdal itself depend on these new dependencies, or just libgdal?

**Comment by gillins** [comment]
Not sure of the point of printing all the OGR driver names. I can remove this if you think it is unnecessary.

**Comment by groutr** [comment]
That was there before I came.  I think it was to verify which drivers OGR has.  However, GDAL's run_test.py does check specifically for the NetCDF driver.  I think we can use that method to check for specific drivers.

File: gdal/run_test.py [hunk 2]
```diff
@@ -10,9 +10,31 @@
...
+ogr.UseExceptions()
+g1 = ogr.Geometry(wkt='POLYGON((0 0, 10 10, 10 0, 0 0))')
+g2 = ogr.Geometry(wkt='POLYGON((-90 -90, -90 90, 190 -90, -90 -90))')
+# following line raises exception if no GEOS
+g1.Contains(g1)
+
+# test for proj.4 support
+osr.UseExceptions()
+sr1 = osr.SpatialReference()
+sr1.ImportFromEPSG(4326) # lat long
```
Comment (groutr) [inline gdal/run_test.py]: `GDAL_DATA` should be set somewhere in this file or this line will fail.

Comment (groutr) [inline gdal/run_test.py]: something like this at the top would work.
```python
import os
os.environ['GDAL_DATA'] = "%s\data" % os.environ['LIBRARY_PREFIX']
```

**Comment by gillins** [comment]
GDAL_DATA should be set on Windows too when libgdal is installed otherwise user's scripts will fail. Is there a way of doing this? I see there is a mechanism for Unix with activate.sh etc.

Is there any documentation for this? I'm guessing a .bat file doing the same thing will work.

**Comment by groutr** [comment]
Good point.  I'm not sure why the original author of the activate scripts didn't make windows versions.  I can work on it in another PR.

**Comment by groutr** [comment]
@gillins, actually on second thought, I think we will have to define GDAL_DATA in the test script.  GDAL_DATA is only defined when activating an environment.  The manner in which conda-build sets up the test environment, the activate/deactivate scripts are never executed.

**Comment by groutr** [comment]
I'll work on getting new builds for these packages in the main repos soon.

**Pull request merged**